src/Stormpath/DataStore/DefaultDataStore.php

```diff
@@ -121,6 +121,12 @@ public function getResource($href, $className, array $options = array())
             $this->addDataToCache($data, $queryString);
         }
 
+        if (isset($options[PropertyBasedClassNameResolver::PROPERTY_ID]))
```

Comment:
This is much better than the previous version, but I still have some change proposals.

The resolver should actually be generic, so this "if" shouldn't be necessary. Inside of the generic resolver, we could determine whether or not it needs the property_id to be set and use the appropriate resolver; otherwise just return the specified class name.